lib/rules/i18n-no-reference.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const { I18N_METHODS } = require('../i18n.js');
+
+function isReferenceNode(node) {
+  if (node.type === 'TemplateLiteral') return false;
+
+  return node.type !== 'Literal' || typeof node.value !== 'string'
+}
+
+function extractI18nMethodAndArgs(node) {
+  let result = [];
+
+  if (node.type === 'CallExpression') {
+    const { callee } = node;
+    if (callee.type === 'MemberExpression') {
+      const { object, property } = callee;
+      if (object.name === 'i18n') {
+        result = [property.name, node.arguments];
+      }
+    }
+  }
+
+  return result;
+}
+
+
+module.exports = {
+  meta: {
+    docs: {
+      category: 'Possible Errors',
+      description:
+        'Ensure predictable static values are passed as i18n method arguments',
+      recommended: false,
+    },
+    fixable: null, // Indicates that some problems reported by this rule are automatically fixable
+    schema: [],
+    messages: {
+      noDynamicValue:
+        'invalid argument type "{{ arg }} passed to i18n.{{ method }}()".',
```

Comment:
so that would print something like this, right?

```
invalid argument type "window passed to i18n.gettext()".
```

I am wondering if we shouldn't quote the argument only, like for `noDynamicValudForSprintf`:

```
invalid argument type "window" passed to i18n.gettext().
```

maybe we should get rid of the "type" in the middle too? so that it would say that there is an invalid argument passed into the i18n method call?

Reply:
It's actually going to pass the AST type, not the literal argument. I could look into seeing if I can get a stringified version of the argument though.

Reply:
Got it working with the literal. Added to both rules and cleaned up the messaging a bit.